vllm/config/model.py

```diff
@@ -437,8 +437,11 @@ def __post_init__(
         skip_mm_profiling: bool | None,
         video_pruning_rate: float | None,
     ) -> None:
+        from vllm.utils.torch_utils import is_torch_equal_or_newer
+
         # Enable batch invariance settings if requested
-        if vllm_is_batch_invariant():
+        # After PyTorch 2.9, no need to default to eager
+        if vllm_is_batch_invariant() and not is_torch_equal_or_newer("2.10.0.dev"):
```

Comment:
Are all of the features only could be used in 2.10? We haven't officially supported it yet.